### ISSUE TITLE
R package uses old dbplyr interface

### ISSUE THREAD

**Issue by hadley** [duckdb/duckdb#3703]
As revealed by using dev version of dbplyr:

``` r
con <- DBI::dbConnect(duckdb::duckdb())
DBI::dbWriteTable(con, "mtcars", mtcars)
mtcars <- dplyr::tbl(con, "mtcars")
#> Warning: <duckdb_connection> uses an old dbplyr interface
#> ℹ Please install a newer version of the package or contact the maintainer
#> This warning is displayed once every 8 hours.
```

<sup>Created on 2022-05-24 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

I have unfortunately forgotten most of what this means, but I'm pretty sure it's a relatively simple fix and you can see the details at https://dbplyr.tidyverse.org/articles/backend-2.html. Let me know if anything isn't obvious and I'll do my best to figure it out.

**Comment by rsund** [comment]
DuckDB uses the dbplyr backend version 2 already. The backend is released in DuckDB v0.3.3, but for some reason that (or newer) version is not yet in CRAN.

**Comment by hannes** [comment]
We have a revdep fail that we still need to figure out with the author of the dependency
